helpers/ipfs.ts

```diff
@@ -12,17 +13,35 @@ type Data = {
   name: string;
   description: string;
   blockchain: string;
+  type: string;
 };
 
-type UploadData = {
-  image: File;
-  name: string;
-  description: string;
+export const uploadToIPFS = async (data: Blob | File) => {
+  if (!data) throw Error("Data is invalid");
+  return await client.storeBlob(data);
 };
 
-export const uploadToIPFS = async (data: UploadData) => {
-  if (!data) throw Error("Data is invalid");
-  return await client.store(data);
+export const pinHashToIPFS = async (ipfsHash: string) => {
+  let CID;
```

Comment:
Minor point but could you extract this into an axios client then import it Just in case it needs to be re-used. Also as an open source project it might be nice to have these slightly deeper abstractions. Something like this:

```typescript
import axios from "axios";

const client = axios.create({
  headers: {
    "Content-Type": "application/json",
    Accept: "application/json",
    Authorization: `Bearer ${process.env.NEXT_PUBLIC_PINATA_JWT_TOKEN}`,
  },
  baseURL: "https://api.pinata.cloud/pinning/pinByHash",
  withCredentials: true,
});

client.interceptors.request.use(
  (config) => {
    return {
      ...config,
      data: {
        // this would have to be set in local storage hashToPin: getIpfsHash,
      },
    };
  },
  function (error) {
    // Do something with request error
    return Promise.reject(error);
  }
);

export default client;
```

Then just use the client in line so:

```javascript
await apiClient.post(config).then().catch();
```

This might be a bit overkill.

Reply:
This is patch fix until I update the subgraph with the new IPFS handling.